src/main/java/com/gluonhq/substrate/config/ConfigResolver.java

```diff
@@ -156,6 +158,21 @@ public ConfigResolver(String classpath) throws IOException, InterruptedException
                 .collect(Collectors.toList());
     }
 
+    /**
+     * Walks through the jars in the classpath,
+     * and looks for META-INF/substrate/config/android-permissions.txt file.
+     *
+     * @return a list of permission lines that should be added to the AndroidManifest file
+     * @throws IOException Exception while reading the permissions file.
+     */
+    public Set<String> getAndroidPermissions() throws IOException {
+        Logger.logDebug("Scanning for android permission files");
+        return Set.copyOf(scanJars(USER_ANDROID_PERMISSIONS_FILE,
```

Comment:
It is probably necessary to return a list/set of permissions that are not duplicated. There is no guarantee that the jars won't contain duplicates, so probably this is a good place to enforce that?

Reply:
`Set` implies uniqueness in case of multiple equal objects.